### PR TITLE
Googletest - rebase to 1.13 

### PR DESCRIPTION
**Pull request by sideb0ard**
Update GoogleTest to version 1.13 to match Chrome branch-heads/5735

b/280671902

### CODE REVIEW

**Comment by andrewsavage1** [comment]
Make sure to preserve and update the METADATA files

Also, why do we have an upstream of external/gtest and external/gmock instead of https://chromium.googlesource.com/chromium/src/testing/? The latter has i.e. the GN code that we need while the former does not

File: base/test/gtest_util.h [hunk 1]
```diff
@@ -33,23 +33,23 @@
...
-    ::testing::internal::RE::PartialMatch(".*", (regex));           \
-    GTEST_SUPPRESS_UNREACHABLE_CODE_WARNING_BELOW_(statement);      \
-    terminator;                                                     \
-  } else                                                            \
-    ::testing::Message()
+// // Macro copied from gtest-death-test-internal.h as it's (1) internal for now
+// // and (2) only defined if !GTEST_HAS_DEATH_TEST which is only a subset of the
+// // conditions in which it's needed here.
+// // TODO(gab): Expose macro in upstream gtest repo for consumers like us that
+// // want more specific death tests and remove this hack.
```
Comment (kaidokert) [inline base/test/gtest_util.h:40]: Nitpick: I'd recommend #if defined(STARBOARD) or OS_STARBOARD or whatever the pattern here is to disable the macro below, rather than block comment. Creates less of a churn

Comment (sideb0ard) [reply base/test/gtest_util.h]: done

**Review by kaidokert — APPROVED** [review]
If things are building, things are looking good to me, lets try and get this ready to land.

We have reviewed this offline too and i think this all is looking good.

File: third_party/googletest/src/googlemock/include/gmock/gmock-nice-strict.h [hunk 2]
```diff
@@ -225,8 +225,11 @@ class GTEST_INTERNAL_EMPTY_BASE_CLASS StrictMock
       "https://google.github.io/googletest/"
       "gmock_cook_book.html#NiceStrictNaggy");
   StrictMock() : MockClass() {
+#if !GTEST_OS_STARBOARD
     static_assert(sizeof(*this) == sizeof(MockClass),
                   "The impl subclass shouldn't introduce any padding");
+#endif // !GTEST_OS_STARBOARD - gives MSVC comp error, i think due to Starboard
```
Comment (kaidokert) [inline third_party/googletest/src/googlemock/include/gmock/gmock-nice-strict.h:231]: I'll add a few nitpicks: This should have at least a follow-up bug logged and in the comment here.

File: third_party/googletest/src/googlemock/src/gmock-cardinalities.cc [hunk 3]
```diff
@@ -33,7 +33,9 @@
 
 #include "gmock/gmock-cardinalities.h"
 
+#if !GTEST_OS_STARBOARD
```
Comment (kaidokert) [inline third_party/googletest/src/googlemock/src/gmock-cardinalities.cc:36]: Ideally we shouldn't be commenting out standard headers, but if this runs into issues with current Starboard, can leave as is.

File: third_party/googletest/src/googlemock/src/gmock-internal-utils.cc [hunk 4]
```diff
@@ -158,13 +162,23 @@ GTEST_API_ void Log(LogSeverity severity, const std::string& message,
 
   if (severity == kWarning) {
     // Prints a GMOCK WARNING marker to make the warnings easily searchable.
+#if !GTEST_OS_STARBOARD
```
Comment (kaidokert) [inline third_party/googletest/src/googlemock/src/gmock-internal-utils.cc:165]: These need a solution for Starboard. Shouldn't we have a `SbLogFormat` here in the #else clause ?

File: third_party/googletest/src/googletest/include/gtest/gtest.h [hunk 5]
```diff
@@ -1349,9 +1349,21 @@ template <typename T1, typename T2>
 AssertionResult CmpHelperEQ(const char* lhs_expression,
                             const char* rhs_expression, const T1& lhs,
                             const T2& rhs) {
+#if defined(_MSC_VER)
```
Comment (kaidokert) [inline third_party/googletest/src/googletest/include/gtest/gtest.h:1352]: A lot of ugly here to silence a warning on a single line. Annoyingly this is in a header too, so moving it into GN code would propagate those snoozed warnings all over the place. I don't know if we have a better solution but eventually this should get cleaned up somehow.

File: third_party/googletest/src/googletest/include/gtest/internal/gtest-internal.h [hunk 6]
```diff
@@ -608,11 +608,11 @@ class GTEST_API_ TypedTestSuitePState {
   bool AddTestName(const char* file, int line, const char* case_name,
                    const char* test_name) {
     if (registered_) {
-      fprintf(stderr,
+      posix::PrintF(
```
Comment (kaidokert) [inline third_party/googletest/src/googletest/include/gtest/internal/gtest-internal.h:611]: I left a comment on a previous review : Would it be possible to do this with a 

```
#define fprintf(a, ...) posix::Printf(__VA_ARGS) 
```

at the top of this file ? Remembering to #undef at the end.

File: third_party/googletest/src/googletest/include/gtest/internal/gtest-port.h [hunk 7]
```diff
@@ -1935,8 +1966,90 @@ inline std::string StripTrailingSpaces(std::string str) {
...
 
+
+#if GTEST_OS_STARBOARD
+typedef int FILE;
+#endif
+
 namespace posix {
 
+#if GTEST_OS_STARBOARD
+
```
Comment (kaidokert) [inline third_party/googletest/src/googletest/include/gtest/internal/gtest-port.h:1977]: This is exactly the type of stuff that i'd like to see completely eliminated with http://go/starboard-16-goals

No action needed here.

File: third_party/googletest/src/googletest/include/gtest/internal/gtest-port.h [hunk 8]
```diff
@@ -2046,6 +2159,7 @@ inline FILE* FDOpen(int fd, const char* mode) { return fdopen(fd, mode); }
 #endif
 inline int FClose(FILE* fp) { return fclose(fp); }
 #if !GTEST_OS_WINDOWS_MOBILE
+
```
Comment (kaidokert) [inline third_party/googletest/src/googletest/include/gtest/internal/gtest-port.h:2162]: Nit: unnecessary

File: third_party/googletest/src/googletest/include/gtest/internal/gtest-port.h (hunk 7 above)
Comment (sideb0ard) [reply third_party/googletest/src/googletest/include/gtest/internal/gtest-port.h:1977]: b/288165638 - clean up bug.

File: third_party/googletest/src/googletest/include/gtest/internal/gtest-internal.h (hunk 6 above)
Comment (sideb0ard) [reply third_party/googletest/src/googletest/include/gtest/internal/gtest-internal.h:611]: Clean up bug - b/288165638

(this was copied over as is from previous rebase)

File: third_party/googletest/src/googletest/include/gtest/gtest.h (hunk 5 above)
Comment (sideb0ard) [reply third_party/googletest/src/googletest/include/gtest/gtest.h:1352]: b/288165638

File: third_party/googletest/src/googlemock/include/gmock/gmock-nice-strict.h (hunk 2 above)
Comment (sideb0ard) [reply third_party/googletest/src/googlemock/include/gmock/gmock-nice-strict.h:231]: Added bug b/288166017

File: third_party/googletest/src/googlemock/src/gmock-cardinalities.cc (hunk 3 above)
Comment (sideb0ard) [reply third_party/googletest/src/googlemock/src/gmock-cardinalities.cc]: removed.

**Comment by kaidokert** [comment]
Just a reminder - i've looked at this several times ( in previous and current form ) - lgtm, approved. Huge CL, if there's some corner case tests failing lets create a follow-up and fix there. Lets try and get it landed because it's such a foundation level part and we have 24 RC date fast approaching.
